nixos/modules/programs/teck-programmer.nix

```diff
@@ -0,0 +1,28 @@
+{ lib, config, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.programs.teck-programmer;
+
+in
+{
+  options.programs.teck-programmer = {
+    enable = mkEnableOption "user access to TECK keyboard firmware";
+    package = mkOption {
+      type = types.package;
+      default = pkgs.teck-programmer;
+      description = "the teck-programmer package";
+    };
+  };
+
+  config = mkIf cfg.enable {
+    environment.systemPackages = [ cfg.package ];
+    services.udev.packages = [
+      (pkgs.runCommand "teck-udev-rules" { } ''
+        mkdir -p $out/etc/udev/rules.d
+        ln -s -t $out/etc/udev/rules.d ${cfg.package}/lib/node_modules/teck-programmer/40-teck.rules
+      '')
+    ];
```

Comment:
I don't know much about `node` packaging in nixpkgs, but can we modify it so `40-teck.rules` is placed in the appropriate folder so all you need is `services.udev.packages = cfg.package;`?

Reply:
not really at the nodePackages stage. However, I'm thinking I should follow the same recipe as `nitrokey-udev-rules` here so I will rewrite this.

Reply:
Thanks for the answer. I'm good either way.